src/components/pagination-nav/pagination-nav.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import settings from '../../globals/js/settings';
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import InitComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import handles from '../../globals/js/mixins/handles';
+
+class PaginationNav extends mixin(createComponent, InitComponentBySearch, handles) {
+  /**
+   * Pagination Nav component
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @extends Handles
+   * @param {HTMLElement} element The element working as a pagination nav.
+   */
+  constructor(element, options) {
+    super(element, options);
+    this.element.addEventListener('click', evt => this.handleClick(evt));
+    this.selectorPageElementArray = this.element.querySelectorAll(this.options.selectorPageElement);
+    this.totalPages = this.selectorPageElementArray.length;
+    this.currentIndex = 0;
+    this.selectorPageDirectionArray = this.element.querySelectorAll(this.options.selectorPageDirection);
+    this.selectorPageSelect = this.element.querySelector(this.options.selectorPageSelect);
+    if (this.selectorPageSelect) {
+      this.selectorPageSelect.addEventListener('change', evt => this.handleSelectChange(evt));
+    }
+  }
+
+  /**
+   * Toggle active state on click
+   */
+  handleClick = evt => {
+    this.selectorPageElementArray.forEach((element, index) => {
+      if (element.classList.contains(this.options.classActive)) {
+        this.currentIndex = index;
```

Comment:
This logic seems to be trying to get the previous index, but does not seem to cope with the selection made in the select box? e.g. Choosing `8` in the select box and hitting left arrow causes moving to `10` instead of `7`.

Reply:
Yeah I wasn't sure what the expected behavior there would be so with the previous next arrows when the select menu is present. I was treating it more like prototype behavior. I'll mature with this in mind!

Reply:
I see - Thank you for clarifying! I think we should update the selection in the select box.